### DISCUSSION TITLE
Add Semitec 104GT-2 Table for Davinci 1.0a models

### DISCUSSION THREAD

**Discussion by zombiRon** [luc-github/Repetier-Firmware-0.92#207] [General]
I've generated a thermistor table for the Semitec 104GT-2 as shipped with the E3D v6 hotend, this covers the original glass and the new cartridge type.

This table is for use with the 1.0a and any other Davinci models that have a single 1kohm resistor instead of the 4k7 and 10k as found on the 1.0 models. I know some users have removed the 10k and used the standard Repetier tables. You could swap out the 1kohm for a 4k7 and use the generic tables, but cut and paste is much easier.

Can this be added to the firmware as an option for users who have upgraded their hotends/thermistors please?

#define EXT0_TEMPSENSOR_TYPE 7
#define NUM_TEMPS_USERTHERMISTOR2 36
#define USER_THERMISTORTABLE2 {{284, 2400},{324, 2320},{371, 2240},{426, 2160},{491, 2080},{566, 2000},{655, 1920},{759, 1840},{880, 1760},{1020, 1680},{1181, 1600},{1363, 1520},{1566, 1440},{1789, 1360},{2028, 1280},{2278, 1200},{2532, 1120},{2781, 1040},{3018, 960},{3234, 880},{3424, 800},{3587, 720},{3719, 640},{3824, 560},{3904, 480},{3964, 400},{4007, 320},{4037, 240},{4057, 160},{4071, 80},{4080, 0},{4085, -80},{4088, -160},{4090, -240},{4091, -320},{4091, -400}}

**Comment by luc-github** [comment]
Thank you , so this table is only valid for 1.0A/2.0A and AiO

Is it a measured table or calculated table ?

**Comment by zombiRon** [comment]
it's a calculated table, I have attempted to verify with a temp probe on my multimeter but as you know multimeters aren't accurate over the range of temps we print at. However I have verified the resistance and voltage at each temp against the Semitec datasheet and these correspond to how the printer interprets the temp.

**Comment by luc-github** [comment]
Ok will add this table in configuration.h with a define for easy selection

**Comment by ZeekWorld** [comment]
Will the table above correct the temperature offset when installing a E3D V6 on a Da Vinci 1.0a?
I have done the upgrade and the temperatures showing on the screen are much higher than what the real temperature is. 
If so, where exactly do I need to paste the code?
Thanks!

**Comment by zombiRon** [comment]
yes this is for use on a 1.0a, if you're using the supplied e3d thermistor. Open configuration.h add it into one of the user thermister tables I think 2 is empty. And define this as the thermistor earlier in the config. If you get stuck I'll post my config. Unless luc has added it as a pre defined (sorry luc I've not reviewed the code for a while)

**Comment by luc-github** [comment]
@zombiRon no it is not yet integrated still added the #Waiting to be processed label - I have been quite busy lately sorry

**Comment by luc-github** [comment]
@ZeekWorld did you managed to use the table ?

**Comment by zombiRon** [comment]
@ZeekWorld, If you look at the code, there is a if MODEL==0 clause before where you have pasted the table

Look a little further down and paste it into the following else section. You have a model 1 device as far as repetier is concerned, so its tying to use an empty table. 

Hope this helps.

**Comment by ZeekWorld** [comment]
@luc-github I am going to give it another shot tonight.
@zombiRon Thank you for the explanation. I believe after trying last night I realized per your message I was missing the #if model==1 you described.

I will flash the Da Vinci tonight and let you both know how it goes.

**Note:** Something I found out after hours of reading and looking for more information is that the original e3d V6 has the Semitec 104GT2 thermistor which @zombiRon created the table for, but I may have purchased a clone of the e3d V6 which most of the threads online refer to a NTC 3950 thermistor with very similar characteristics. So I am not sure if having a different thermistor will make any difference even both are 100k.

**Comment by zombiRon** [comment]
This table will not work with the NTC thermistor, the tables are very specific to each thermistor. Table 5 is the closest you'll get to the generic NTC thermistors that come with the clone hot ends. 

Regards,
Aaron

**Comment by ZeekWorld** [comment]
Thanks Aaron.

**Comment by reklawdiov** [comment]
I just installed my new style e3d v6 hotend with the cartridge style thermistor, I got it from an authorized dealer so it's not a clone. I used @zombiRon 's custom table and when I set the extruder temp in the LCD menu to 265c it never gets there, it stops at around 257c. it's definitely hot enough to feed and melt ABS all day long but I'm not convinced it's reporting the correct temp and I can't verify with my temp gun because it wigs out with reflections. Am I missing a setting? PID tuning? 

where are these resistors that I need to check for 1k or 4k7 or 10k? I'm proficient with soldering so If I can bump off or replace them to use the standard table that's cool. Also, this is a 1.0A printer.

**Comment by zombiRon** [comment]
Hi reklawdiov,
The 1k internal resistance on the 1.0A is not as easily accessible as it was on the 1.0 version. I attempted to track it down, after hours of following traces I couldn't find a simple way of physically changing the internal resistance. However, while building the custom table I realised that the 1K makes more sense from a calculation point of view. I think your issue is down to software configuration. Can you post your config.h please? But just to be sure, disconnect the thermistor and measure the resistance on the printers thermistor pins.

**Comment by reklawdiov** [comment]
@zombiRon  I'll check with the multimeter when I get home after work but here is my config.h file, thanks for checking!
[Configuration.zip](https://github.com/luc-github/Repetier-Firmware-0.92/files/1363153/Configuration.zip)

**Comment by zombiRon** [comment]
Your config is good, only other thing I can think of is your EEPROM settings, here are the relevant settings from my EEPROM...

Temp Ext PLA: 180.000 
Temp Ext ABS: 220.000 
Temp Bed PLA: 60.000 
Temp Bed ABS: 90.000
Extr.1 heat manager [0-3]: 1
Extr.1 PID drive max: 255
Extr.1 PID drive min: 40
Extr.1 PID P-gain/dead-time: 7.0300 
Extr.1 PID I-gain: 0.7000 
Extr.1 PID D-gain: 17.6900 
Extr.1 PID max value [0-255]: 255
Extr.1 temp. stabilize time [s]: 0
Extr.1 temp. for retraction when heating [C]: 200
Extr.1 distance to retract when heating [mm]: 0
Extr.1 extruder cooler speed [0-255]: 255
Extr.1 advance L [0=off]: 0.000

**Comment by reklawdiov** [comment]
@zombiRon that seems to have done it! I finally got a chance to test and tweak today, it was using "slowbang" heat manager 3 and the PIDs were way off. I've updated and am able to hit 270C without a problem (so far) other than the menu maxes out at 270 instead of 280 that's set as max in the printer settings. probably for the best since it says not to go above 285 I think for the thermistor.

**Comment by jnowak1054** [comment]
Please excuse my ignorance as I'm probably not thinking clearly and missing something simple. Currently running Repetier 92.10 on a davinci 1.0a. I've been attempting to build a proper table for my unknown thermister in my current hotend as it reads way low (reads 280C @ 240C as measured by a thermocouple.) 
When I change line 365 in Configuration.h 
#define EXT0_TEMPSENSOR_TYPE 5 
to #define EXT0_TEMPSENSOR_TYPE 7
the printer reports back 0C
I tried pasting a new calculated table into #define USER_THERMISTORTABLE0 on line 723, #define EXT0_TEMPSENSOR_TYPE 5, with no change in reported temperature from the printer in original config.
Compared the above config.h to my own and see line 721 also has #if MODEL==0, and tried changing that anyway as well. Tried along with different combinations of EEPROM Extr.0-3 heat manager [0-3]:
I'm obviously missing something. Any help or a point in the right direction to look or read would be greatly appreciated.

**Comment by luc-github** [comment]
@jnowak1054 you have a 1.0a so your model must be MODEL 1
keep `#define EXT0_TEMPSENSOR_TYPE 5`

you just need to modify the lines : 
https://github.com/luc-github/Repetier-Firmware-4-Davinci/blob/master/src/ArduinoDUE/Repetier/Configuration.h#L737-L738

**Reply by 1RonRock** [reply]
> 
> 
> @jnowak1054 you have a 1.0a so your model must be MODEL 1
> keep `#define EXT0_TEMPSENSOR_TYPE 5`
> 
> you just need to modify the lines :
> https://github.com/luc-github/Repetier-Firmware-4-Davinci/blob/master/src/ArduinoDUE/Repetier/Configuration.h#L737-L738

What do I need to modify? 

I have set my Model at 1, define EXT0_TEMPSENSOR_TYPE 5.

Upload to printer. Bed goes to 100 as set temp. Nozzle goes to 260 then decouples. reset and decouples at 237. Setpoint is 285.

I tried opening the above Config zip file to check. When I unzip I get a Configuration.h file Arduino will not open it so I am unable to see that configuration.

**Reply by 1RonRock** [reply]
I was able to open the zip file in wordpad. It is quite different than what I have in my config tables.

**Comment by jnowak1054** [comment]
Thanks! Now that I see it, I also see my question was already answered above. Thanks again!

**Comment by zombiRon** [comment]
This reminds me, I've not been active for a while; luc, your corexy code works too. I converted my davinci to a custom corexy. Happy to share the drawings if anyone is interested.

**Comment by sisensee** [comment]
Always interested!

**Comment by jnowak1054** [comment]
@zombiRon I'll second sisensee's interest, would like to see what you've done.

**Comment by zombiRon** [comment]
Here's the mod on Thingiverse https://www.thingiverse.com/thing:4492584

**Comment by 1RonRock** [comment]
I am missing something here.

I copied the table given by zombiRon in first post to my table2 (line #735 in Command h) I first deleted the existing table.
I also changed thermister type to 36 as in his post.

I then changed my EXT0_TEMPSENSOR_TYPE to 7 as explained.

Saved and uploaded the configuration.

When I try to connect Repetier to use printer I get an fatel/error. I seem to be able to connnect but the fatel/temp sensor error will not let me proceed.

**Reply by zombiRon** [reply]
You are using a genuine Semitec thermistor?

**Reply by 1RonRock** [reply]
I believe so. I bought a genuine E3D-V6.

**Reply by zombiRon** [reply]
ok, start with multimeter and make sure thermistor is reading correct resistance at room temp when not connected to printer. You can find the semitec data sheet via everyone's favourite search engine.

**Reply by 1RonRock** [reply]
Thank you. I will do that.

**Comment by 1RonRock** [comment]
Pretty sure that it is reading correct. The datasheet I found on Semitec shows a very large range of temps. It shows,

104NT-4

10*c  208.8 kΩ
25*c 100 kΩ


My ambient temp =20.5C  reading 113.2 kΩ

For some reason E3D does not have a datasheet. They do sell a replacement thermister Semitec 104Gt. Their page says,   "Cartridge type thermistors use a Semitec 104NT thermistor, potted in a cylindrical stainless steel cartridge with a high conductivity ceramic based adhesive. "

So I found the datasheet on 104 NT.

**Reply by zombiRon** [reply]
that looks good, so now measure the internal resistance of the printer. With the printer OFF, measure the resistance between the terminals where you would connect the thermistor to.
If you don't get 10K or 4K7, then you have loose/shorted wiring; fix that then we can look at the software.

**Comment by 1RonRock** [comment]
1.091kΩ at the thermister connector.

**Reply by 1RonRock** [reply]
So does this indicate an issue in the wiring/board? Seems odd I was able to print yesterday until I loaded the temp table. Not saying that my temps were correct, just that it did work.

Now it seems as though my printer has hit a "hard stop" I can not load or unload filament, it never goes to the temp rising.

**Reply by 1RonRock** [reply]
I traced the wires from the thermister connector to the small board. The connectors on the small board to determine the correct wires going to the motherboard. Black and orange wires. All wires test good. continuity through the pin connectors on the small board. I get nearly the same reading at the pins on the motherboard 1.027kΩ.

Now I'm lost. Again.....

**Reply by 1RonRock** [reply]
After the last post I went back up and read this,

"This table is for use with the 1.0a and any other Davinci models that have a single 1kohm resistor instead of the 4k7 and 10k as found on the 1.0 models. "

Does that indicate that I am getting correct resistance readings after all?

**Reply by zombiRon** [reply]
Hardware is fine, and this confirms it's a 1.0a model. This means it must be configuration. I suspect the problem is WHERE the patch is inserted is the issue.

**Reply by 1RonRock** [reply]
I agree.

**Comment by 1RonRock** [comment]
I set the EXT0_TEMPSENSOR_TYPE to 5 and reloaded the configuration. Now I am able to heat the nozzle again. But the 5 does not correspond to the correct table. 

I also made changes to the EEPROM settings through Repetier to the settings refereed above.

**Reply by zombiRon** [reply]
can you post your configuration.h file please? or if your using Git clone of the repo you can git diff and post the patch.

**Reply by 1RonRock** [reply]
Thank you, I have tried to copy your configuration h from above. In that effort I found that the table you provide is in a different location than I expected. So I tried to make that location as well as where I thought it should be.  Here is my Configuration h that does not work,


[Configuration.zip](https://github.com/luc-github/Repetier-Firmware-4-Davinci/files/6217366/Configuration.zip)

**Reply by zombiRon** [reply]
This looks correct, with the tables inserted into lines 745+746 for a model=1 which is the davinci 1.0a with J37 and 1K resistor.

What are you using for compilation? Can you clean and rebuild the binary?

**Reply by zombiRon** [reply]
wait, I think I found your problem...
Line 2160:
`#define UI_SET_MIN_HEATED_BED_TEMP  60`
Your bed is triggering the thermistor issue

Actually, you have modified more extreme values throughout, please set these back to the defaults.

**Reply by 1RonRock** [reply]
I'm not certain what you are asking, but I "think" this is it.

I am using Arduino 1.8.10

Across the  top is File, Edit, Sketch, Tools, Help.

Under Sketch I have several choices, Two that seem like they might do what you ask are Verify/Compile, and Export Compiled Binary

Under Tools one of the choices is Fix Encoding & Reload.

I have not tried any of these since I don't know what they do.

**Reply by zombiRon** [reply]
I stopped using Arduino IDE long ago, but it doesn't always clean and recompile which means changes can be ignored. A quick google suggests closing the Arduino IDE and reopening the project will force it to clean the cache. Maybe try this.

**Reply by 1RonRock** [reply]
I think that the changes you have found are from the EEPROM settings listed above. I will try to find defaults.

**Reply by zombiRon** [reply]
EEPROM cannot change the source code, in your configuration.h on line 2160, change this back to 30 and try that as a first step.

**Reply by 1RonRock** [reply]
I closed Arduino, then reopened.

I reset that value to 30.

Uploaded to printer. 

Close Arduino, open Repiter. Connect to printer. Fatel/Error.

**Reply by zombiRon** [reply]
Just to be clear you are seeing the following behaviour...
1. Make changes to configuration.h
2. Build and Upload with Arduino IDE
3. Printer resets after upload and reports Fatal Error

**Reply by 1RonRock** [reply]
Sorry I replied to this yesterday from my phone, for some reason it does not seem to have went through.

Yes. After I upload, then start the printer I go to EEPROM Settings and load Fail Safe. As instructed by LUC. Then open Repetier and when I connect to the printer through Repetier I get the Fatal Error.

I have found that this only seems to happen when I have set my  EXT0_TEMPSENSOR_TYPE to 7. If I change it to 5 the printer seems ok. No error, but also not correct temp table.

**Reply by zombiRon** [reply]
before you connect repetier host, does it display correct temp on the printer display after it resets?

**Reply by 1RonRock** [reply]
The printer has been powered off for several hours. So should be pretty close to ambient temp.

On the DaVinci screen it shows the nozzle at 17C bed at 19C

My Fluke multimeter shows 19.6C with theFluks temp thermister near the bed. Nozzle is in the clean out area.

So not perfect, but pretty close to each other. 

I wonder if there is a formatting error I have done when copying your table? Maybe I left an extra space? Wrong type font? IDK just guessing.

**Reply by zombiRon** [reply]
this is with the thermistor set to type 7??

**Reply by 1RonRock** [reply]
It is at 5 right now. I will set it to 7 and report back.

**Reply by 1RonRock** [reply]
I changed EXT0_TEMPSENSOR_TYPE from 5 to 7. Upload config. No other changes. When printer restarts after upload, the screen on it shows def in the nozzle temp position. I did not go any further.

I should add that I do see 20 in the bed temp on the screen. Correct temp.

**Reply by zombiRon** [reply]
Def is expected when the printer measures a disconnected or faulty thermistor. This also happens with wrong or missing temp table. I'll upload a config.h for you to copy in and compile against. I suspect the copy paste has introduced smart quotes.

**Reply by 1RonRock** [reply]
Cool. Not sure what "smart quotes" are but I do think that is likely the problem. Only wish that I was smart enough to figure it out.

**Reply by zombiRon** [reply]
[Configuration.h.zip](https://github.com/luc-github/Repetier-Firmware-4-Davinci/files/6228761/Configuration.h.zip)
unzip and copy this file over your existing configuration.h do not edit it, then build/upload to printer with Arduino IDE

**Reply by 1RonRock** [reply]
Well crap! 

After I copy the file that you just provide, then compile to check I get an error in the Arduino. Very long message but bottom line is,

'LANGUAGE_JP_ACTIVE' was not declared in this scope

Again I am lost. I did try to search but nothing that I understand.

[error.txt](https://github.com/luc-github/Repetier-Firmware-4-Davinci/files/6229656/error.txt)

**Reply by 1RonRock** [reply]
I downloaded and installed the current version of Arduino on a different computer. Loaded the config files with the Configuration that you provided. Same error.

I thought that the error may have been from my old Arduino software and the messing around that I have done.

Line 1293 in this view. Not sure what tab, none of them are highlighted.

**Reply by zombiRon** [reply]
this is my mistake sorry, please find a revised version 
[Configuration.h.zip](https://github.com/luc-github/Repetier-Firmware-4-Davinci/files/6230470/Configuration.h.zip)

**Reply by 1RonRock** [reply]
Well it seems to work. I had a hard time getting the com port to work after the upload to the printer. For some reason Windows would not recognize the port on restart. After several tries it worked.

I see that it is set at EXT0_TEMPSENSOR_TYPE 5. Is that correct or should it be at 7? 

The nozzle does heat, runs up to 269-270 then decouples. I can reset it from the error message and it heats again, but decouples if I let it get back to 269-270. My max temp is set at 290, in my testing I have the manual temp set at 285.

Also noticed that the nozzle temp was at 2 on first start after upload. Bed temp was 20 (correct)

**Reply by 1RonRock** [reply]
I think it is working!

I changed the EXT0_TEMPSENSOR_TYPE to 7.

Made the changes in the EEPROM settings from the Repetier config to match the settings shown above in this thread.

The nozzle goes up to and holds 285. At least it did once.

**Reply by zombiRon** [reply]
If you scroll to the bottom of the file you'll see there's an override. This was just a test config, I wouldn't use it full time.

But now you know it works you can copy paste (using Arduino IDE not word or wordpad they introduce smart quotes) into a fresh copy of the source and make a permanent config.

**Reply by 1RonRock** [reply]
I'll have to try to figure that out. At least I have a config that I am certain works. I'll save it and should be able to always get back if needed.

I did google the smart quotes you mentioned. Learned a little, not enough to fix the issue.  I am very impressed with the time and patience that you have put into helping me with this. It's very frustrating for me to know that there is a problem, have an idea what it is, but not be able to fix it. I'll compare the working config that you just sent to one of the non working ones and try to see the difference.

**Reply by 1RonRock** [reply]
Well, I opened the test config that you provided in Arduino. Then opened the config that is refereed to in the thread by Luc. Copied the table from the test config to the  NUM_TEMPS_USERTHERMISTOR2 36. changed to EXT0_TEMPSENSOR_TYPE to 7 above. Loaded config to my printer. Same results as previously. Very frustrating! 

There seems to be something missing, earlier in the thread Luc mentions to leave EXT0_TEMPSENSOR_TYPE to 5 and make changes to lines 737, & 738. But no explanation what to change them to. Most confusing is why changing EXT0_TEMPSENSOR_TYPE to 7 does not work.

Here is my newly made config if interested,


[Config zip.zip](https://github.com/luc-github/Repetier-Firmware-4-Davinci/files/6251162/Config.zip.zip)

**Reply by zombiRon** [reply]
[Configuration.h.zip](https://github.com/luc-github/Repetier-Firmware-4-Davinci/files/6254642/Configuration.h.zip)
You put the usertable in the wrong section. Here is a corrected version

**Reply by 1RonRock** [reply]
YAY! You got it working. Again thank you very much.
